go/vumitools/app_worker.py

```
@@ -378,8 +366,34 @@ def add_conv_to_msg_options(self, conv, msg_options):
 
 
 class GoApplicationMixin(GoWorkerMixin):
-    # TODO: Move some stuff to here.
-    pass
+    def get_config_for_conversation(self, conversation):
+        config_data = GoApplicationConfigData(self.config, conversation)
+        return self.CONFIG_CLASS(config_data)
+
+    @inlineCallbacks
+    def get_message_config(self, msg):
+        if isinstance(msg, TransportEvent):
+            msg = yield self.find_message_for_event(msg)
+
+        msg_mdh = self.get_metadata_helper(msg)
+        conversation = yield msg_mdh.get_conversation()
+
+        returnValue(self.get_config_for_conversation(conversation))
+
+
+class GoRouterMixin(GoWorkerMixin):
+    def get_config_for_router(self, router):
+        config_data = GoApplicationConfigData(self.config, router)
+        return self.CONFIG_CLASS(config_data)
+
+    @inlineCallbacks
+    def get_message_config(self, msg):
+        # By the time we get an event here, the metadata has already been
+        # populated for us from the original message.
```

Might be good to say what has populated the metadata? I assume it's meant to be the routing table router? Perhaps we should raise a suitable error if its not populated? Can the same change be made for GoApplicationMixin?